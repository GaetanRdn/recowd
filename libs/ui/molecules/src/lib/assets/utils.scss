@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
