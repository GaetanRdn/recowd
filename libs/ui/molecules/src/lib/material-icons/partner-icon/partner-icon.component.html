<rc-material-icon type="outlined">handshake</rc-material-icon>
