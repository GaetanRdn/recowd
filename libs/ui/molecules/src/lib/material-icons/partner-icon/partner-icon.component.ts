import { ChangeDetectionStrategy, Component } from '@angular/core';
import { MaterialIconDirective } from '@recowd/ui-atoms';

@Component({
  selector: 'rc-partner-icon',
  standalone: true,
  imports: [MaterialIconDirective],
  template: ` <rc-material-icon type="outlined">handshake</rc-material-icon>`,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class PartnerIconComponent {}
