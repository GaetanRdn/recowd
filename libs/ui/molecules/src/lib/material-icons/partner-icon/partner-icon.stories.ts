import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { PartnerIconComponent } from './partner-icon.component';

export default {
  title: 'molecules/material-icons/partner-icon',
  parameters: {
    jest: 'partner-icon',
  },
  decorators: [
    moduleMetadata({
      imports: [PartnerIconComponent],
    }),
  ],
} as Meta<PartnerIconComponent>;

const Template: Story<PartnerIconComponent> = () => ({
  template: `<rc-partner-icon></rc-partner-icon>`,
});

export const Default = Template.bind({});
