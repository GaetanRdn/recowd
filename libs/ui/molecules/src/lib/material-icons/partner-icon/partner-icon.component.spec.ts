import { TestBed, waitForAsync } from '@angular/core/testing';

import { PartnerIconComponent } from './partner-icon.component';
import { TemplateLookup } from '@recowd/test/utils';

describe('PartnerIconComponent', () => {
  let templateLookup: TemplateLookup<PartnerIconComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      imports: [PartnerIconComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(PartnerIconComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.firstChildElement).toMatchSnapshot());
});
