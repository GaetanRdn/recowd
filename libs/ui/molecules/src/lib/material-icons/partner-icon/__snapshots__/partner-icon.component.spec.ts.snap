// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartnerIconComponent should create 1`] = `
<rc-material-icon
  class="material-icons-outlined"
  type="outlined"
>
  handshake
</rc-material-icon>
`;
